packages/mui-material/src/Select/SelectInput.js

```diff
@@ -405,7 +425,10 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
         }
       },
       role: 'option',
-      selected,
+      selected:
+        arr[0].props.value === undefined || arr[0].props.disabled === true
```

Comment:
@michaldudak @DouglasPds 

This assumes that the first child is always a valid React component.
This leads to crash if the first child is of different type. Here is a first [report](https://github.com/mui/material-ui/issues/32253)

Reply:
Yes, I tried here and when the first child is a number, string, NaN, it crashes. The first solution that works was to put optional chaining to check the values after the props. Here where the value is passed to selected and inside the function isFirstSelectableElement.

Reply:
`arr[0].props?.value === undefined || arr[0].props?.disabled === true`

You are suggesting this, right? As long as this doesn't break the logic of the fix this PR originally contained, I am fine with that. Will you be willing to ensure that and open a PR with the change?